internal/controllers/nmc_reconciler.go

```diff
@@ -377,10 +445,18 @@ func (w *workerHelperImpl) SyncStatus(ctx context.Context, nmcObj *kmmv1beta1.No
 
 		switch phase {
 		case v1.PodFailed:
-			status.InProgress = false
 			updateModuleStatus = true
 			podsToDelete = append(podsToDelete, p)
 
+			// If we failed to load, we can just remove the status.
```

Comment:
I think this code should not be here: SyncStatus should only update the statuses based on the pods. The decision regarding status removal is done in the ProcessOrphanModuleStatus. No need to spread it across the code.

Reply:
`ProcessOrphanModuleStatus` cannot determine if a status should be removed after an unloading Pod was successful. We do it here because we know the Pod's action (in the `actionLabelKey` label).

Reply:
But this code is dealing with failed Load pod, and it already handled in the ProcessOrphanModuleStatus